src/devices/machine/mc68328.h

```diff
@@ -920,10 +809,437 @@ class mc68328_device : public m68000_device
+	u8  pkdata_r();
+	u8  pkpuen_r();
+	u8  pksel_r();
+	u8  pmdata_r();
+	u8  pmdir_r();
+	u8  pmpuen_r();
+	u8  pmsel_r();
+
+	virtual u16 pwmc_r() override;
+	u16 pwmp_r();
+	u16 pwmw_r();
+	u16 pwmcnt_r();
+
+	u16 wctlr_r();
+	u16 wcmpr_r();
+	u16 wcn_r();
+
+	u16 spisr_r();
+
+	u8  llbar_r();
+	u8  lotcr_r();
+	u16 lgpmr_r();
+
+	// $(FF)FFF108
+	u16 m_grpmaska;     // Chip Select Group A Mask Register
+	u16 m_grpmaskb;     // Chip Select Group B Mask Register
+	u16 m_grpmaskc;     // Chip Select Group C Mask Register
+	u16 m_grpmaskd;     // Chip Select Group D Mask Register
+	u32 m_csa[4];       // Group A Chip Select Registers
+	u32 m_csb[4];       // Group B Chip Select Registers
+	u32 m_csc[4];       // Group C Chip Select Registers
+	u32 m_csd[4];       // Group D Chip Select Registers
+
+	// $(FF)FFF300
+	u32 m_iwr;          // Interrupt Wakeup Enable Register
+
+	// $(FF)FFF438
+	u8  m_pjdir;        // Port J Direction Register
+	u8  m_pjdata;       // Port J Data Register
+	u8  m_pjsel;        // Port J Select Register
+
+	u8  m_pkdir;        // Port K Direction Register
+	u8  m_pkdata;       // Port K Data Register
+	u8  m_pkpuen;       // Port K Pullup Enable Register
+	u8  m_pksel;        // Port K Select Register
+
+	u8  m_pmdir;        // Port M Direction Register
+	u8  m_pmdata;       // Port M Data Register
+	u8  m_pmpuen;       // Port M Pullup Enable Register
+	u8  m_pmsel;        // Port M Select Register
+
+	// $(FF)FFF502
+	u16 m_pwmp;         // PWM Period Register
+	u16 m_pwmw;         // PWM Width Register
+	u16 m_pwmcnt;       // PWM Counter
+
+	// $(FF)FFF600
+	timer_regs m_timer_regs[2];
+	u16 m_wctlr;        // Watchdog Control Register
+	u16 m_wcmpr;        // Watchdog Compare Register
+	u16 m_wcn;          // Watchdog Counter
+
+	// $(FF)FFF700
+	u16 m_spisr;        // SPIS Register
+
+	void internal_map(address_map &map);
+	void cpu_space_map(address_map &map);
+
+	// internal state
+	virtual void register_state_save() override;
+
+	virtual u32 get_irq_mask_for_level(int level) override;
+	virtual int get_irq_level_for_mask(u32 level) override;
+
+	virtual TIMER_CALLBACK_MEMBER(pwm_tick) override;
+
+	virtual emu_timer *get_timer(int timer) override;
+	virtual timer_regs &get_timer_regs(int timer) override;
+	virtual u32 get_timer_int(int timer) override;
+
+	virtual u16 lcd_get_lxmax_mask() override;
+	virtual u32 lcd_get_line_word_count() override;
+	virtual attotime lcd_get_line_rate() override;
+	virtual u8 lcd_get_panel_bit_size() override;
+	virtual attotime get_pixclk_rate() override;
+
+	virtual bool rtc_int_is_active() override;
+	virtual u16 rtc_get_int_mask() override;
+	virtual bool rtc_get_alarm_match() override;
+
+	emu_timer   *m_gptimer[2];
+
+	devcb_write_line    m_out_port_j_cb[8];
+	devcb_write_line    m_out_port_k_cb[8];
+	devcb_write_line    m_out_port_m_cb[8];
+
+	devcb_read_line     m_in_port_j_cb[8];
+	devcb_read_line     m_in_port_k_cb[8];
+	devcb_read_line     m_in_port_m_cb[8];
```

Comment:
These are candidates for `devcb_write_line::array<8>` and `devcb_read_line::array<8>` as well.